src/stdlib_experimental_stats.md

```diff
@@ -47,3 +48,55 @@ program demo_mean
                    reshape(x, [ 2, 3 ] ) > 3.)  !returns [ NaN, 4.0, 5.5 ]
 end program demo_mean
 ```
+
+## `var` - variance of array elements
+
+### Description
+
+Returns the variance of all the elements of `array`, or of the elements of `array` along dimension `dim` if provided, and if the corresponding element in `mask` is `true`.
+
+The variance is defined as the best unbiased estimator and is computed as:
+
+```
+ var(x) = 1/(n-1) sum_i (array(i) - mean(array))^2
+```
+
+### Syntax
+
+`result = var(array [, mask])`
+
+`result = var(array, dim [, mask])`
+
+### Arguments
+
+`array`: Shall be an array of type `integer`, `real`, or `complex`.
+
+`dim`: Shall be a scalar of type `integer` with a value in the range from 1 to n, where n is the rank of `array`.
+
+`mask` (optional): Shall be of type `logical` and either by a scalar or an array of the same shape as `array`.
+
+### Return value
+
+If `array` is of type `real` or `complex`, the result is of the same type as `array`.
+If `array` is of type `integer`, the result is of type `double precision`.
```

Comment:
@nshaffer Thank you for your review.

I used `double precision` because they are declared as `dp`. But I agree they are actually `real(real64)`. The issue with using `real64` in the spec is that if the definition of `dp` 
 in `stdlib_experimental_kinds` changes (there has been already discussions on that), then we will need to modify the spec too.

Would it be better to write "..... the result is of type `dp`."?